pandas/core/arrays/interval.py

```diff
@@ -913,7 +913,7 @@ def isna(self) -> np.ndarray:
         return isna(self._left)
 
     def shift(
-        self: IntervalArrayT, periods: int = 1, fill_value: object = None
+        self: IntervalArray, periods: int = 1, fill_value: object = None
```

Comment:
I removed the TypeVar for `self`.

I'm not going to not mark this as "resolved" so that other people can more easily stumble upon this and maybe investigate it further :)